src/hub_api/client.py

```diff
@@ -167,25 +163,44 @@ async def _variant_details(  # noqa: PLR0911
             case _:  # pragma: no cover
                 assert_never(variant.plugin.plugin_type)
 
+    async def find_plugin(
+        self,
+        *,
+        plugin_name: str,
+        plugin_type: enums.PluginTypeEnum | None = None,
+        variant_name: str | None = None,
+    ) -> api_schemas.PluginDetails:
+        conditions: list[sa.ColumnElement[bool]] = [models.Plugin.name == plugin_name]
+        if plugin_type is not None:
+            conditions.append(models.Plugin.plugin_type == plugin_type)
+        if variant_name is not None:
+            conditions.append(models.PluginVariant.name == variant_name)
+        else:
+            conditions.append(models.PluginVariant.id == models.Plugin.default_variant_id)
+
+        q = sa.select(models.PluginVariant).join(models.Plugin, sa.and_(*conditions))
+
+        # TODO: Handle ambiguity:
+        # - Plugins of different type with the same name
+        if variant := (await self.db.execute(q)).scalar():
```

Comment:
Trying to get more than one row here to start handling ambiguity forces me to use `.unique()` on the result, so I've got some reading to do: 

- https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#orm-rows-not-uniquified-by-default
- https://github.com/sqlalchemy/sqlalchemy/discussions/11770
- https://github.com/sqlalchemy/sqlalchemy/issues/4395
- https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html#joined-eager-loading